### ISSUE TITLE
Errors occur when trying to run the website locally

### ISSUE THREAD

**Issue by jihoon-seo** [kubernetes-sigs/prow#25]
I tried to run the website locally, but some errors occurred.

[env]
- npm version: 8.5.5
- node version: v16.15.0
- Hugo version: v0.91.2+extended (the same as https://github.com/kubernetes-sigs/prow/blob/main/netlify.toml#L8)
- Go version: 1.16.5 (the same as https://app.netlify.com/sites/k8s-prow/deploys/6360082776136c00088f311e#L38)

[prereq]
- `make init-theme`
- `make update-theme`

[cmd & output]
```
❯ hugo server -v                                     
Start building sites … 
hugo v0.91.2+extended linux/amd64 BuildDate=unknown
INFO 2022/11/11 16:15:24 syncing static files to /
ERROR 2022/11/11 16:15:24 render of "page" failed: execute of template failed: template: docs/single.html:30:7: executing "docs/single.html" at <partial "scripts.html" .>: error calling partial: "/home/jhseo/prow/site/themes/docsy/layouts/partials/scripts.html:67:107": execute of template failed: template: partials/scripts.html:67:107: executing "partials/scripts.html" at <resources.Concat>: error calling Concat: slice []interface {} not supported in concat
ERROR 2022/11/11 16:15:24 render of "page" failed: execute of template failed: template: docs/single.html:30:7: executing "docs/single.html" at <partial "scripts.html" .>: error calling partial: "/home/jhseo/prow/site/themes/docsy/layouts/partials/scripts.html:67:107": execute of template failed: template: partials/scripts.html:67:107: executing "partials/scripts.html" at <resources.Concat>: error calling Concat: slice []interface {} not supported in concat
ERROR 2022/11/11 16:15:24 render of "page" failed: execute of template failed: template: docs/single.html:30:7: executing "docs/single.html" at <partial "scripts.html" .>: error calling partial: "/home/jhseo/prow/site/themes/docsy/layouts/partials/scripts.html:67:107": execute of template failed: template: partials/scripts.html:67:107: executing "partials/scripts.html" at <resources.Concat>: error calling Concat: slice []interface {} not supported in concat
ERROR 2022/11/11 16:15:24 render of "page" failed: execute of template failed: template: docs/single.html:30:7: executing "docs/single.html" at <partial "scripts.html" .>: error calling partial: "/home/jhseo/prow/site/themes/docsy/layouts/partials/scripts.html:67:107": execute of template failed: template: partials/scripts.html:67:107: executing "partials/scripts.html" at <resources.Concat>: error calling Concat: slice []interface {} not supported in concat
ERROR 2022/11/11 16:15:24 failed to render pages: render of "page" failed: execute of template failed: template: docs/single.html:30:7: executing "docs/single.html" at <partial "scripts.html" .>: error calling partial: "/home/jhseo/prow/site/themes/docsy/layouts/partials/scripts.html:67:107": execute of template failed: template: partials/scripts.html:67:107: executing "partials/scripts.html" at <resources.Concat>: error calling Concat: slice []interface {} not supported in concat
Error: Error building site: TOCSS: failed to transform "scss/main.scss" (text/x-scss): SCSS processing failed: file "stdin", line 6, col 1: File to import not found or unreadable: ../vendor/bootstrap/scss/bootstrap. 
Built in 169 ms
```

**Comment by jihoon-seo** [comment]
Downgrading Docsy to v0.2.0 resolved this issue.

[How to downgrade Docsy to v0.2.0]
- `cd site/themes/docsy`
- `git pull`
- `git checkout tags/v0.2.0`
- `git submodule update --init --recursive --depth 1`
- `cd ../..` (I mean, `prow/site/`)

---

- It seems that it would be great to specify the Docsy version in the `.gitmodules` file.
- In future, we might bump Docsy to newer than `v0.2.0`.

**Comment by jihoon-seo** [comment]
First of all, I cannot figure out & want to know which Docsy version is used in the Netlify production/preview builds.

---

Although I wrote [this comment](https://github.com/kubernetes-sigs/prow/issues/25#issuecomment-1311381288), it seems that downgrading Docsy to v0.3.0 is not a sustainable & desirable solution.

One noticeable point is that Docsy (v0.3.0 → v0.4.0) changed the way to import Bootstrap and FontAwesome - from 'git submodules' to NPM packages, and this might break Netlify production/preview builds and/or local preview builds if Docsy ~v0.3.0 is used in Netlify builds for now and we are to bump the Docsy version.
- https://github.com/google/docsy/issues/948
- https://github.com/google/docsy/pull/889

Currently Docsy is suggesting to [convert an existing site to use Docsy as a Hugo Module](https://www.docsy.dev/docs/updating/convert-site-to-module/),
and adopting this conversion to this repo might resolve this issue somehow magically.

**Comment by listx** [comment]
I can try converting us to use Docsy as a Hugo Module to try to see if that helps things. As a user though I think I'm more interested in solving the broken icons issue you filed in #27. Maybe using a Hugo Module will fix both issues at once?

**Comment by jihoon-seo** [comment]
@listx Honestly I am not sure whether:
- using Hugo Module (and bumping Docsy to newer version) will resolve this issue, and
- using Hugo Module will resolve the issue #27.

IMO it would be great for us to
- apply the suggestion and see whether we can run website locally from clean env without error, and
- apply the suggestion to prod (because #27 is not reproducible on previews and local) and see whether 
  - Netlify build for prod site succeeds without error, and
  - icons are displayed well.

One more thing is that we might update some deps as well when applying the fix, like:
- Hugo (ref: https://github.com/kubernetes/website/pull/34356)
- ...

**Comment by listx** [comment]
> apply the suggestion

By this do you mean using Docsy as a Hugo Module or something else?

**Comment by jihoon-seo** [comment]
Yes I meant 'using Docsy as a Hugo Module'. 😊

**Comment by listx** [comment]
See #35.

**Comment by jihoon-seo** [comment]
This issue seems to be resolved by the PR #35.

---

I checked and confirm that

on
- A new AWS EC2 VM
- Install
  - Go 1.19.4
  - Hugo v0.91.2+extended (the same as https://github.com/kubernetes-sigs/prow/blob/main/netlify.toml#L8)

User can run website locally by just running `make` which executes the default Makefile rule (`serve`),
without preparing prereqs by like:
- `make init-theme`
- `make update-theme`
- `git submodule update`
- `npm install`

So I am going to close this issue.
Please feel free to comment or reopen if anyone needs to do so.
